lib/asset/assetmetadata.js

```diff
@@ -29,13 +30,13 @@ class AssetMetadata {
      */
     constructor(filename, contentType, contentLength) {
         if (filename && typeof filename !== "string") {
-            throw Error(`Invalid filename: ${filename}`);
+            throw new IllegalArgumentError("Invalid filename", filename);
```

Comment:
```suggestion
            throw new IllegalArgumentError("Invalid 'filename'", filename);
```
I think we try usually to put variable names in quotes in error/log messages (to better distinguish them from text).